internal/dialect/dialectquery/ydb.go

```diff
@@ -41,10 +41,6 @@ func (c *Ydb) GetMigrationByVersion(tableName string) string {
 }
 
 func (c *Ydb) ListMigrations(tableName string) string {
-	// "--!syntax_pg" enables query processing with PostgreSQL-compatible syntax.
-	// YQL by design strictly forbids the execution of SELECT statements without columns from ORDER BY clause.
-	// In PostgreSQL-compatible mode, SELECT statements can be processed without columns from ORDER BY clause.
-	q := `--!syntax_pg
-	SELECT version_id, is_applied FROM %s ORDER BY tstamp DESC`
+	q := `SELECT version_id, is_applied FROM %s ORDER BY version_id DESC`
```

Comment:
The sort here should either be id or timestamp, but not version.

In the future, this may change, but for now, this query should return a list of migrations in applied order.

Reply:
Is version_id not ordered by migrations?

Reply:
Clickhouse have the same query
https://github.com/pressly/goose/blob/master/internal/dialect/dialectquery/clickhouse.go#L37

Reply:
By default, yes. But, it's not guaranteed because goose supports out-of-order migrations.

https://pressly.github.io/goose/blog/2021/out-of-order-migrations/

Reply:
ClickHouse is a bit of a special case, but I also consider that a bug.

The only time this matters is if you're opting into out-of-order migrations and applying down migrations. If you have migrations that were applied <1s apart then we cannot guarantee deterministic ordering.

A bit more context on this here:

https://github.com/pressly/goose/pull/520#discussion_r1196538231

Most users do not use out-of-order and so this is a non-issue for them.

Reply:
I suggest approve this PR.
And after fix issue https://github.com/ydb-platform/ydb/issues/1350 (maybe 24.1 - 24Q1-24Q2) I will contribute another PR with ordering by tstamp

Reply:
I suppose that's the least-worst option, better to have an edge case as opposed to a non-functioning implementation.

Again, this is really only an issue for down migrations when allowing out-of-order.

Reply:
Alternate way - re-write query to
```sql
SELECT version_id, is_applied FROM (
    SELECT version_id, is_applied, tstamp 
    FROM goose_db_version ORDER BY tstamp DESC
)
```
This is working query, but it has no garanties for ordering in bounded SELECT
The order is preserved right now, but it may break down in the future